boards/arm/nucleo_h743zi/pinmux.c

```diff
@@ -22,9 +22,9 @@ static const struct pin_config pinconf[] = {
 	{ STM32_PIN_PB8, STM32H7_PINMUX_FUNC_PB8_I2C1_SCL },
 	{ STM32_PIN_PB9, STM32H7_PINMUX_FUNC_PB9_I2C1_SDA },
 #endif
-#ifdef CONFIG_PWM_STM32_12
+#if DT_HAS_NODE(DT_NODELABEL(timers12))
```

Comment:
This needs a better solution (timer is now only used for PWM, but that may change). I'd like to hear proposals, maybe labeling each PWM with `pwm{n}`?

Reply:
I have submitted a PR for solving a similar issue for the NXP FlexTimer (which is currently only used for PWM in Zephyr, but has many uses - some, which I am working on providing drivers for) in #24498. My proposed solution is similar to that used for handling the different use-cases for the NXP Flexcomm instances in Zephyr.

Reply:
Thanks @henrikbrixandersen. It looks like a good approach that can be applied here. If I get it right the functionality is selected by specifying which driver (`compatible`) will use that peripheral.

Reply:
FYI: This is how it is handled on U-Boot (same in Kernel):
https://elixir.bootlin.com/u-boot/latest/source/arch/arm/dts/stm32mp157c.dtsi#L132
https://elixir.bootlin.com/u-boot/latest/source/arch/arm/dts/stm32mp157c-ev1.dts#L300.

Reply:
> If I get it right the functionality is selected by specifying which driver (`compatible`) will use that peripheral.

@gmarull That is correct.

Reply:
@henrikbrixandersen This article may be useful as it describes in detail how Linux handles the same problem https://bootlin.com/blog/timer-counters-linux-microchip/ (what a coincidence, it was posted today :smile: )

Reply:
@gmarull on my side, I'd prefer to go with the label addition as I think was your initial proposal (same as what is describe in the article you mention finaly). This  would help to stay close to stm32 linux.